src/Core/Content/Test/Seo/SeoUrlGeneratorTest.php

```diff
@@ -194,6 +202,90 @@ public function testVariantInheritance(): void
         static::assertEquals('parent/green', $urls[$ids->get('green')]);
     }
 
+    public function testNotBeingStateful(): void
+    {
+        $categoryIds = $this->getCategoryIds(2);
+
+        self::assertCount(2, $categoryIds, 'this is important for the test as you need more items to iterate for a context switch test');
+
+        /** @var \Generator $firstRun */
```

Comment:
can you type it more explict what it is 😅

Reply:
Do I have to? It just needs to be a generator (what it really is) to be a traversable for later test usage.